### PR TITLE
chore: add action that comments on feature request label event

### PR DESCRIPTION
**Pull request by atierian**
*Issue #, if available:*
N/A

*Description of changes:*
Adds GitHub Action that posts the following message to an issue after it is labeled `feature-request`.
> This has been identified as a feature request. If this feature is important to you, we strongly encourage you to give a 👍 reaction on the request. The feedback we receive from the community is a large factor in helping us prioritize new features.

*Check points: (check or cross out if not relevant)*

- [ ] ~Added new tests to cover change, if needed~
- [ ] ~Build succeeds with all target using Swift Package Manager~
- [ ] ~All unit tests pass~
- [ ] ~All integration tests pass~
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/feature_request.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  issues: write
+
+jobs:
+  project:
+    name: Feature request message
+    runs-on: ubuntu-latest
+    if: ${{ github.event.label.name == 'feature-request' }}
+
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
```
Comment (ameter) [inline .github/workflows/feature_request.yml:17]:
```suggestion
```
optimization: you should be able to specify the repo in the gh cli command and not need to perform the checkout (in conjunction with other suggestion below)

File: .github/workflows/feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - name: add feature request comment
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ISSUE_NUMBER: ${{ github.event.issue.number }}
+        run: |
+          gh issue comment $ISSUE_NUMBER -b "This has been identified as a feature request. If this feature is important to you, we strongly encourage you to give a 👍 reaction the request. The feedback we receive from the community is a large factor in helping us prioritize new features."
```
Comment (ameter) [inline .github/workflows/feature_request.yml:24]:
```suggestion
          gh issue comment $ISSUE_NUMBER --repo aws-amplify/amplify-ios -b "This has been identified as a feature request. If this feature is important to you, we strongly encourage you to give a 👍 reaction the request. The feedback we receive from the community is a large factor in helping us prioritize new features."
```

**Review by ameter — APPROVED** [review]
Approved with suggestion.

File: .github/workflows/feature_request.yml (hunk 1 above)
Comment (atierian) [reply .github/workflows/feature_request.yml:17]: Nice! TIL

**Review by ameter — APPROVED** [review]
Approved with minor suggestion.  Also, please check with PM on verbiage before merging.

**Review by hdworld11 — APPROVED** [review]

**Review by ameter — APPROVED** [review]

**Pull request merged**